### PR TITLE
add initial travis config

### PR DESCRIPTION
**Pull request by coreysprague**
add Travis CI to the repo.

Note, I had to temporarily disable the golden assertions for non Mac platforms for now. I will file an issue to add support for platform specific Goldens so that we can rely on the Mac ones for local development and linux for CI.

### CODE REVIEW

File: packages/golden_toolkit/pubspec.yaml [hunk 1]
```diff
@@ -16,3 +16,8 @@ dependencies:
 
 dev_dependencies:
   pedantic: ^1.8.0
+
+# needed so the images can be loaded in the example tests
```
Comment (coreysprague) [inline packages/golden_toolkit/pubspec.yaml]: I over-zealously removed this in my last PR and CI didn't exist to catch it!

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+before_script:
+  - git clone https://github.com/flutter/flutter.git -b stable --depth 1
+  - ./flutter/bin/flutter doctor
+  - ./flutter/bin/flutter packages get
+script:
+  - cd packages/golden_toolkit
+  - ../../flutter/bin/flutter analyze
+  - ../../flutter/bin/cache/dart-sdk/bin/dartfmt -n ./lib --set-exit-if-changed
+  - ../../flutter/bin/flutter test --coverage --coverage-path=lcov.info
+# after_success:
```
Comment (mZadorskii) [inline .travis.yml]:
```suggestion
```

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+#   - bash <(curl -s https://codecov.io/bash)
```
Comment (mZadorskii) [inline .travis.yml]:
```suggestion
```

Comment (coreysprague) [reply .travis.yml]: actually... better yet...

File: packages/golden_toolkit/test/golden_builder_test.dart [hunk 4]
```diff
@@ -35,7 +37,11 @@ Future<void> main() async {
         wrapper: noWrap(),
       );
 
-      await screenMatchesGolden(tester, 'square_container');
+      await screenMatchesGolden(
+        tester,
+        'square_container',
+        skip: !Platform.isMacOS,
```
Comment (mZadorskii) [inline packages/golden_toolkit/test/golden_builder_test.dart]: I'm expecting to see a comment here about why Platform.isMacOS

File: .travis.yml (hunk 3 above)
Comment (coreysprague) [reply .travis.yml]: ebay teams are already using codecov. Just going to turn it on 😄

**Pull request merged**